geoviews/util.py

```diff
@@ -580,28 +579,37 @@ def process_crs(crs):
     """
     try:
         import cartopy.crs as ccrs
-        import geoviews as gv # noqa
+        import pyproj
     except ImportError:
-        raise ImportError('Geographic projection support requires GeoViews and cartopy.')
+        raise ImportError('Geographic projection support requires geoviews, pyproj and cartopy.')
 
     if crs is None:
         return ccrs.PlateCarree()
 
-    if isinstance(crs, str) and crs.lower().startswith('epsg'):
+    errors = []
+    if isinstance(crs, str):
+        starts = ['epsg:', '+init=epsg:']
+        for start in starts:
+            if crs.lower().startswith(start):
+                try:
+                    return ccrs.epsg(crs[len(start):].strip())
+                except Exception as e:
+                    errors.append(e)
```

Comment:
If I'm understanding correctly, it's stripping the non-digits?

If so, I think this is more readable.

```suggestion
    if isinstance(crs, str):
        try:
            return ccrs.epsg("".join(c for c in crs if c.isdigit()))
        except Exception as e:
            errors.append(e)
```

Or if ccrs.epsg does accept non-digits:
`crs.lower().replace("epsg:", "").replace("+init=", "")`

Reply:
I had the same idea but also didn't know if `epsg` did accept non-digits. 

Looking at https://en.wikipedia.org/wiki/EPSG_Geodetic_Parameter_Dataset it seems to be numbers between 1024 and 32767.

Reply:
A small note is that it is faster to use list comprehension than a generator, see https://github.com/adamchainz/flake8-comprehensions/issues/156#issuecomment-1475149702.

Reply:
Yeah I think it's only codes:
https://scitools.org.uk/cartopy/docs/latest/_modules/cartopy/crs.html#epsg

Reply:
It is very negligible, but we should take what we can :upside_down_face: